### PR TITLE
Simplify caching and fetching network information

### PR DESCRIPTION
**Pull request by pravisankar**
Precursor for storing/fetching plugin name from ClusterNetwork.

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/networking PTAL

**Comment by danwinship** [comment]
LGTM

File: plugins/osdn/common.go [hunk 1]
```diff
@@ -140,64 +140,64 @@ func (oc *OsdnController) validateNetworkConfig(clusterNetwork, serviceNetwork *
...
 		return true, nil
 	}
 	return false, nil
 }
 
 func (oc *OsdnController) StartMaster(clusterNetworkCIDR string, clusterBitsPerSubnet uint, serviceNetworkCIDR string) error {
 	// Validate command-line/config parameters
-	hostBitsPerSubnet := int(clusterBitsPerSubnet)
-	clusterNetwork, _, serviceNetwork, err := ValidateClusterNetwork(clusterNetworkCIDR, hostBitsPerSubnet, serviceNetworkCIDR)
+	network, err := ValidateClusterNetwork(clusterNetworkCIDR, int(clusterBitsPerSubnet), serviceNetworkCIDR)
```
Comment (dcbw) [inline plugins/osdn/common.go]: Maybe 'network' should be 'ni' here to be consistent with other parts of the code?  I won't block a merge for that though.

File: plugins/osdn/registry.go [hunk 2]
```diff
@@ -126,75 +132,63 @@ func (registry *Registry) GetPod(nodeName, namespace, podName string) (*kapi.Pod
...
-	}
-	return registry.clusterNetwork, registry.hostSubnetLength, registry.serviceNetwork, nil
-}
-
-func (registry *Registry) GetClusterNetwork() (*net.IPNet, error) {
-	if err := registry.cacheClusterNetwork(); err != nil {
 		return nil, err
 	}
-	return registry.clusterNetwork, nil
+	return ValidateClusterNetwork(cn.Network, cn.HostSubnetLength, cn.ServiceNetwork)
```
Comment (dcbw) [inline plugins/osdn/registry.go]: Shouldn't this cache the return value of ValidateClusterNetwork() into registry.NetworkInfo?  I can't find where the cached object gets set, but I'm probably missing it while skimming the patch.

**Comment by pravisankar** [comment]
@dcbw @danwinship Updated, PTAL

**Comment by dcbw** [comment]
LGTM now

**Pull request merged**